src/appshell/view/welcomedialogmodel.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-Studio-CLA-applies
+ *
+ * MuseScore Studio
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2025 MuseScore Limited
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "welcomedialogmodel.h"
+#include "log.h"
+
+using namespace mu::appshell;
+
+WelcomeDialogModel::WelcomeDialogModel()
+{
+}
+
+void WelcomeDialogModel::init()
+{
+    // TODO: This is entirely placeholder code - figure out where to store/pull this information...
```

Comment:
Let’s move this to StartupScenario and make WelcomeDialog more abstract - that way, if we decide to fetch data from the server in the future, part of the work will already be in place, and the dialog will be ready for it.